nl_server/embeddings.py

```diff
@@ -80,52 +131,201 @@ def get_embedding_at_index(self, index: int) -> List[float]:
   def get_embedding(self, query: str) -> List[float]:
     return self.model.encode(query).tolist()
 
-  def detect_svs(self, query: str) -> Dict[str, Union[Dict, List]]:
-    query_embeddings = self.model.encode([query])
+  #
+  # Given a list of queries, searches the in-memory embeddings index
+  # and returns a map of candidates keyed by input queries.
+  #
+  def _search_embeddings(self, queries: List[str]) -> Dict[str, VarCandidates]:
```

Comment:
Question: why do the refactor of this function in the same PR as the introduction of a new function for detecting multi svs? Is there some functionality that was missing which this refactor addresses?

Reply:
The refactor takes multiple queries and returns results, which was needed for the multi SV case.